src/main/java/org/terasology/navgraph/BaseRegion.java

```diff
@@ -20,30 +20,52 @@
 
 /**
  * @author synopia
+ * Abstract class which floors and regions derive from
  */
 public abstract class BaseRegion<N extends BaseRegion> {
     public int id;
     public Set<N> neighborRegions = new HashSet<N>();
     protected BitMap map;
     protected Floor floor;
 
+    /**
+     * Creates a BaseRegion
+     * @param id the id of the base region
+     */
     protected BaseRegion(int id) {
         this.id = id;
         map = new BitMap();
     }
 
+    /**
+     * Marks a point in the region as walkable by characters
+     * @param x the x co-ordinate of the point with respect to region's top-left corner
```

Comment:
- what is the "top-left corner" of a region?
- what defines the sizes of a region?
- what exactly does "walkable" mean, e.g., is this about solid ground characters can walk on, or about _passable_ blocks characters can move through (might be swimming, crouching, ...)?

Reply:
So they way it was created, walkable blocks are all those blocks on which a (1x1x1) cube character can walk.  A non penetrable block underneath and some air blocks above. (Currently set to 2 IIRC). The don't include any other features such as swimming/ crouching.